### PR TITLE
PP-1307: Test "test_sister_mom_crash" of TestSisterMom Fails as it is not able to find pbsdsh path while submitting the job

### PR DESCRIPTION
**Pull request by Mahalty**
#### Bug/feature Description
* *[PP-1307](https://pbspro.atlassian.net/browse/PP-1307)*

#### Analysis
*  In the test we are submitting job as pbsuser with contents of the job script as "pbsdsh dd if=/dev/zero of=/dev/null". On the systems where the test failed, the test wasn't able to fetch the path for the command "pbsdsh", hence job didn't execute successfully.

#### Solution Description
* Test need to be updated with the path of pbsdsh command while submitting the job

#### Testing logs/output
* [Execution_logs_PP-1307_after_fix.txt](https://github.com/PBSPro/pbspro/files/2419499/Execution_logs_PP-1307_after_fix.txt)
* [Execution_logs_PP-1307_without_fix.txt](https://github.com/PBSPro/pbspro/files/2419500/Execution_logs_PP-1307_without_fix.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by Mahalty** [comment]
Thanks @VishrutArora 
@PBSPro/pbspro-maintainers Could any one of you please review this PR

File: test/tests/functional/pbs_sister_mom_crash.py [hunk 1]
```diff
@@ -53,7 +53,8 @@ def test_sister_mom_crash(self):
             self.skipTest("test requires atleast two MoMs as input, " +
                           "use -p moms=<mom1:mom2>")
         sister_mom = self.moms.keys()[1]
-        script = "pbsdsh dd if=/dev/zero of=/dev/null"
+        pbsdsh_path = self.du.which(exe='pbsdsh')
+        script = "%s dd if=/dev/zero of=/dev/null" % pbsdsh_path
```
Comment (bhroam) [inline test/tests/functional/pbs_sister_mom_crash.py]: The problem with which() is that it mainly looks through the path to find a program.  We know where pbsdsh is.  Just join pbs_conf['PBS_EXEC'] with bin and pbsdsh

**Comment by Mahalty** [comment]
Thanks @bhroam for the review. I have addressed your comments. Please have a look
[Latest_Execution_logs_PP-1307.txt](https://github.com/PBSPro/pbspro/files/2432976/Latest_Execution_logs_PP-1307.txt)

File: test/tests/functional/pbs_sister_mom_crash.py [hunk 2]
```diff
@@ -53,7 +53,9 @@ def test_sister_mom_crash(self):
             self.skipTest("test requires atleast two MoMs as input, " +
                           "use -p moms=<mom1:mom2>")
         sister_mom = self.moms.keys()[1]
-        pbsdsh_path = self.du.which(exe='pbsdsh')
+        conf = self.du.parse_pbs_config()
+        pbs_exec = conf['PBS_EXEC']
+        pbsdsh_path = os.path.join(pbs_exec, "bin", "pbsdsh")
```
Comment (bhroam) [inline test/tests/functional/pbs_sister_mom_crash.py]: I don't think you need to parse pbs.conf.  I think self.server.pbs_conf['PBS_EXEC']

**Comment by Mahalty** [comment]
Latest execution logs:
[Latest_Execution_logs_1_PP-1307.txt](https://github.com/PBSPro/pbspro/files/2440944/Latest_Execution_logs_1_PP-1307.txt)

**Comment by Mahalty** [comment]
Thanks @bhroam for the review.

@PBSPro/pbspro-maintainers: I have squashed my commits and rebased my branch. Could any one of you please take this PR forward

**Pull request merged**